src/script/mod.rs

```diff
@@ -178,100 +298,65 @@ impl Code<'_> {
-                    })
-                })
-                .and_then(|final_state| {
-                    if !final_state.vexec().is_empty() {
-                        Err(Error::UnclosedConditional(final_state.vexec().len()))
-                    } else {
-                        Ok(final_state.stack)
-                    }
-                })
-        } else {
-            Err(Error::ScriptSize(Some(self.0.len())))
-        }
-    }
-
     /// Pre-version-0.6, Bitcoin always counted CHECKMULTISIGs
     /// as 20 sigops. With pay-to-script-hash, that changed:
     /// CHECKMULTISIGs serialized in script_sigs are
     /// counted more accurately, assuming they are of the form
     ///  ... OP_N CHECKMULTISIG ...
     pub fn get_sig_op_count(&self, accurate: bool) -> u32 {
-        let parser = self.parse();
-        match iter::once(Ok(None))
-            .chain(parser.map(|r| r.map(Some)))
-            .zip(parser)
-            .try_fold(0, |n, ops| match ops {
-                (Ok(last_opcode), Ok(opcode)) => Ok(n + match opcode {
-                    opcode::PossiblyBad::Good(Opcode::Operation(op)) => match op {
-                        OP_CHECKSIG | OP_CHECKSIGVERIFY => 1,
-                        OP_CHECKMULTISIG | OP_CHECKMULTISIGVERIFY => match last_opcode {
-                            Some(opcode::PossiblyBad::Good(Opcode::PushValue(
-                                opcode::PushValue::SmallValue(sv),
-                            )))
-                                // Even with an accurate count, 0 keys is counted as 20 for some
-                                // reason.
-                                if accurate && OP_1 <= sv => {
-                                    u32::try_from(sv.to_num()).expect("`sv` is positive")
-                                }
-                            // Apparently it’s too much work to figure out if it’s one of the few
-                            // `LargeValue`s that’s valid, so we assume the worst.
-                            Some(_) => u32::from(interpreter::MAX_PUBKEY_COUNT),
-                            // We’re at the beginning of the script pubkey, so any pubkey count must
-                            // be part of the script sig – assume the worst.
-                            None => u32::from(interpreter::MAX_PUBKEY_COUNT),
-                        },
-                        _ => 0,
-                    },
-                    _ => 0,
-                }),
-                (_, _) => Err(n),
-            }) {
-            Err(n) => n,
-            Ok(n) => n,
+        iter::get_sig_op_count(self.parse(), accurate)
+    }
+}
+
+impl Evaluable for Code {
+    /// Get the byte length of this script sig.
+    fn byte_len(&self) -> usize {
+        self.0.len()
+    }
+
+    /// Convert a sequence of `Opcode`s to the bytes that would be included in a transaction.
+    fn to_bytes(&self) -> Vec<u8> {
+        self.0.to_vec()
```

Comment:
After the change to make `Code` own its bytes, this would be clearer:
```suggestion
        self.0.clone()
```